google/cloud/pubsub/internal/default_pull_ack_handler_test.cc

```diff
@@ -83,8 +86,20 @@ Status PermanentError() {
 
 TEST(PullAckHandlerTest, AckSimple) {
   auto subscription = pubsub::Subscription("test-project", "test-subscription");
+  // Start lease loop past max extension so that we don't get any modify ack
```

Comment:
Not sure if this is the best way to mock the clock. I tried putting it in a function, but then the calls weren't being properly mocked. Open to suggestions.

Reply:
https://github.com/googleapis/google-cloud-cpp/blob/main/google/cloud/internal/clock.h

Then your tests can use:

https://github.com/googleapis/google-cloud-cpp/blob/6a938abba9bb5336b393cc8cd462dec2dba2196d/google/cloud/bigtable/internal/rate_limiter_test.cc#L29

Reply:
Good suggestion. I changed this to a refactoring PR. I will add the lease management in the next PR.